### PR TITLE
Run rapids-dependency-file-generator via pre-commit

### PR DESCRIPTION
**Pull request by vyasr**
## Description

This feature is enabled by the most recent release of dfg. This changes also allows us to disable the dfg CI check as it is now redundant.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/rmm/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by vyasr** [comment]
Converting to draft pending some discussions on https://github.com/rapidsai/shared-action-workflows/issues/46.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
 repos:
     - repo: https://github.com/pre-commit/pre-commit-hooks
       rev: v4.3.0
       hooks:
           - id: trailing-whitespace
           - id: end-of-file-fixer
+    - repo: https://github.com/rapidsai/dependency-file-generator
+      rev: v1.2.0
+      hooks:
+          - id: rapids-dependency-file-generator
```
Comment (vyasr) [inline .pre-commit-config.yaml:11]:
```suggestion
          - id: rapids-dependency-file-generator
          - args: ["--clean"]
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,10 +1,14 @@
-# Copyright (c) 2022, NVIDIA CORPORATION.
+# Copyright (c) 2022-2023, NVIDIA CORPORATION.
 repos:
     - repo: https://github.com/pre-commit/pre-commit-hooks
       rev: v4.3.0
       hooks:
           - id: trailing-whitespace
           - id: end-of-file-fixer
+    - repo: https://github.com/rapidsai/dependency-file-generator
+      rev: v1.2.0
```
Comment (vyasr) [inline .pre-commit-config.yaml:9]:
```suggestion
      rev: v1.4.0
```

**Review by ajschmidt8 — APPROVED** [review]

**Pull request merged**